modules/bmc/ipmiscanner.rb

```diff
@@ -0,0 +1,82 @@
+require 'bmc/basescanner'
+require 'concurrent'
+require 'bmc/bmc_plugin'
+
+module Proxy
+  module BMC
+    class IPMIScanner < BaseScanner
+      include Proxy::Log
+      include Proxy::Util
+
+      # This is an IPMI ping, not an ICMP ping.
+      def address_pings?(address)
+        begin
+          socket = UDPSocket.new
+        rescue Errno::EMFILE
+          logger.warn "IPMIScanner: Ran out of free file descriptors while creating UDPSocket! Consider increasing file open limit."
+          retry
```

Comment:
This has a potential to loop for while. I'd suggest putting a hard limit on number of retries and handle file descriptor starvation by reducing the number of simultaneously contacted addresses instead.